### ISSUE TITLE
Dotted Indent Line 

### ISSUE THREAD

**Issue by barslmn** [jupyter/notebook#2837]
Hi,
Is there an option for dotted indent line.

Thanks

Example: https://github.com/Yggdroot/indentLine

**Comment by jasongrout** [comment]
No, I don't think anyone has implemented that. It would be a Codemirror thing if implemented. One codemirror post from 2013 indicated a possible way forward if someone wanted to implement such a thing: https://groups.google.com/forum/#!topic/codemirror/Jud58Iu9QsU

**Comment by juhasch** [comment]
If you are OK with codemirror rulers:
https://codemirror.net/demo/rulers.html

you could try the ruler notebook extension:
http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/ruler/readme.html

**Comment by eric-wieser** [comment]
Some progress seems to have been made here: https://discuss.codemirror.net/t/indentation-guides-design-advice/1119/2

**Comment by eric-wieser** [comment]
This looks like all that is needed:

https://github.com/wmertens/brackets-indent-guides/blob/master/main.js#L50-L106

**Comment by riyadhrazzaq** [comment]
@eric-wieser any idea how to use this- https://github.com/wmertens/brackets-indent-guides/blob/master/main.js#L50-L106 for Jupyter Notebook? not sure where to paste and such.

**Comment by eric-wieser** [comment]
@riyadhrazzaq: No idea. My assumption was that someone who has touched the jupyter CodeMirror integration would be able to work out where to put that. Somewhere jupyter has an equivalent to the `codeMirror` object from that snippet.
